stablehlo/dialect/TypeInference.cpp

```diff
@@ -3625,15 +3619,17 @@ LogicalResult verifyReduceOp(std::optional<Location> location,
   SmallVector<ShapedType> initValueTypes{llvm::map_range(
       initValues.getTypes(), [](Type t) { return t.cast<ShapedType>(); })};
 
-  // P1. & P2.
+  if (dimensions.getType().getRank() != 1)
+    return emitOptionalError(location, "dimensions must be rank 1");
```

Comment:
Also, how about pushing this code into `verifyReduceOpInputsAndInferShape`? (As well as updating the comment on all callers of verifyReduceOpInputsAndInferShape to include reduce_i3).